pkg/backends/cloudwatch/cloudwatch_test.go

```diff
@@ -17,6 +18,22 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// testContext returns a context that will timeout and fail the test if not canceled.  Used to
+// enforce a timeout on tests.  Copied from fixtures_test.go.
+func testContext(t *testing.T) (context.Context, func()) {
+	ctxTest, completeTest := context.WithTimeout(context.Background(), 1100*time.Millisecond)
+	go func() {
+		after := time.NewTimer(1 * time.Second)
+		select {
+		case <-ctxTest.Done():
+			after.Stop()
+		case <-after.C:
+			require.Fail(t, "test timed out")
```

Comment:
I think this is a data race. Please see https://golang.org/pkg/testing/#T and https://github.com/golang/go/issues/15758 and https://github.com/golang/go/issues/24680

Reply:
An approach to achieve the same result may be to move everything else into goroutines and block on timer in the test goroutine somehow. Then if it fires fail the test. This will not work if the other code also needs to be able to do the same. =\

Reply:
Well bugger. This pattern isn't new in the code base. I'll raise a new issue to refactor it out of the project. It "works" enough for now though, are you ok with leaving this instance in?

Reply:
Yes, np.